src/packages/client/scripts/postinstall.js

```diff
@@ -241,56 +241,82 @@ function getPostInstallTrigger() {
   try {
     npm_config_argv = JSON.parse(maybe_npm_config_argv_string)
   } catch (e) {
-    return (
-      UNABLE_TO_FIND_POSTINSTALL_TRIGGER_JSON_PARSE_ERROR +
-      `: ${maybe_npm_config_argv_string}`
-    )
+    return `${UNABLE_TO_FIND_POSTINSTALL_TRIGGER_JSON_PARSE_ERROR}: ${maybe_npm_config_argv_string}`
   }
 
   if (typeof npm_config_argv !== 'object' || npm_config_argv === null) {
-    return (
-      UNABLE_TO_FIND_POSTINSTALL_TRIGGER_JSON_SCHEMA_ERROR +
-      `: ${maybe_npm_config_argv_string}`
-    )
+    return `${UNABLE_TO_FIND_POSTINSTALL_TRIGGER_JSON_SCHEMA_ERROR}: ${maybe_npm_config_argv_string}`
   }
 
   const npm_config_arv_original_arr = npm_config_argv.original
 
   if (!Array.isArray(npm_config_arv_original_arr)) {
-    return (
-      UNABLE_TO_FIND_POSTINSTALL_TRIGGER_JSON_SCHEMA_ERROR +
-      `: ${maybe_npm_config_argv_string}`
-    )
+    return `${UNABLE_TO_FIND_POSTINSTALL_TRIGGER_JSON_SCHEMA_ERROR}: ${maybe_npm_config_argv_string}`
   }
 
   const npm_config_arv_original = npm_config_arv_original_arr
     .filter((arg) => arg !== '')
     .join(' ')
 
-  if (npm_config_arv_original === '') {
-    return (
-      UNABLE_TO_FIND_POSTINSTALL_TRIGGER__EMPTY_STRING +
-      `: ${maybe_npm_config_argv_string}`
-    )
+  const command = `${[getPackageManagerName(), npm_config_arv_original]
+    .filter((arg) => arg !== '')
+    .join(' ')}`
```

Comment:
I'm confused, here it looks like the filter is applied to the array `[getPackageManagerName(), npm_config_arv_original]`, it should only be applied to `npm_config_arv_original` no? but filtering is also done line 257.

Also is the template literal necessary, it makes it hard to read and could be removed I think.

Reply:
> Also is the template literal necessary, it makes it hard to read and could be removed I think.

Remnants of some old code; Will fix!

Reply:
> but filtering is also done line 257.

Because the result of `npm_config_arv_original` can be an empty string. I have however adjusted the code to be less list-oriented and handle this one-off case in a one-off way. I hope it's clearer now. Let me know!